src/main/java/org/jenkinsci/plugins/workflow/steps/StepDescriptor.java

```diff
@@ -109,9 +193,33 @@ public Step newInstance(Map<String,Object> arguments) throws Exception {
      * @param step a fully-configured step (assignable to {@link #clazz})
      * @return arguments that could be passed to {@link #newInstance} to create a similar step instance
      * @throws UnsupportedOperationException if this descriptor lacks the ability to do such a calculation
+     * @deprecated
+     *      Use {@link #uninstantiate(Step)}
      */
     public Map<String,Object> defineArguments(Step step) throws UnsupportedOperationException {
-        return DescribableModel.uninstantiate_(step);
+        if (Util.isOverridden(StepDescriptor.class, getClass(), "uninstantiate", Step.class)) {
+            // if the subtype has defined the uninstantiate() method, delegate to that
+            return uninstantiate(step).toMap();
```

Comment:
This is not really going to work, is it? Because it will be returning a map with entries containing `UninstantiatedDescribable` values; whereas callers of this method will be expecting entries containing `Map` values.